src/Giraffe/Core.fs

```diff
@@ -435,8 +435,7 @@ module Core =
     /// <param name="str">The string value to be send back to the client.</param>
     /// <returns>A Giraffe <see cref="HttpHandler" /> function which can be composed into a bigger web application.</returns>
     let setBodyFromString (str: string) : HttpHandler =
-        let bytes = Encoding.UTF8.GetBytes str
-        fun (_: HttpFunc) (ctx: HttpContext) -> ctx.WriteBytesAsync bytes
+        fun (_: HttpFunc) (ctx: HttpContext) -> ctx.WriteStringAsync str
```

Comment:
> ...because now it constructs a new dictionary on each next-call. Will there be more next-calls than `routePorts` calls? Just to note that Dictionary is not thread-safe so not to put it outside `let routePorts`.

We might need to check the IL/ASM and run a benchmark to check if there's any noticeable performance improvement. Furthermore, if we make this change, I'd like to add new automated tests to assert things are still working in the long run.